build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -136,6 +136,15 @@ function(iree_bytecode_module)
     get_filename_component(_FRIENDLY_NAME "${_RULE_SRC}" NAME)
   endif()
 
+  set(_DEPENDS "")
+  iree_package_name(_PACKAGE_NAME)
+  list(TRANSFORM _RULE_DEPENDS REPLACE "^::" "${_PACKAGE_NAME}::")
+  foreach(_DEPEND ${_RULE_DEPENDS})
+    string(REPLACE "::" "_" _DEPEND "${_DEPEND}")
+    list(APPEND _DEPENDS ${_DEPEND})
+  endforeach()
```

Comment:
This code is strangely mixing `::`-separated namespaces with `_`-separated package names.

I think you want `iree_package_ns` to get a `::`-separated namespace, like this:

```cmake
  iree_package_ns(_PACKAGE_NS)
  list(TRANSFORM _RULE_DEPENDS REPLACE "^::" "${_PACKAGE_NS}::")
```

Now you have proper namespaces that are consistently `::`-separated, not half `::` and half `_`, so this should work as-is, you should be able to drop the `foreach` loop here which IIUC was only needed to finish converting to `_`-separated as we already had the `_` from the package name.